app/com/linkedin/drelephant/tuning/hbt/ParameterGenerateManagerHBT.java

```diff
@@ -0,0 +1,257 @@
+  protected List<JobSuggestedParamSet> getPendingParamSets() {
+    List<JobSuggestedParamSet> pendingParamSetList = _executionEngine.getPendingJobs()
+        .eq(JobSuggestedParamSet.TABLE.tuningAlgorithm + "." + TuningAlgorithm.TABLE.optimizationAlgo,
+            TuningAlgorithm.OptimizationAlgo.HBT.name())
+        // .eq(JobSuggestedParamSet.TABLE.isParamSetDefault, 0)
+        .findList();
+    logger.debug(
+        " Number of Pending Jobs for parameter suggestion " + this._executionEngine + " " + pendingParamSetList.size());
+    return pendingParamSetList;
+  }
+
+  @Override
+  protected List<TuningJobDefinition> getTuningJobDefinitions() {
+    List<TuningJobDefinition> totalJobs = _executionEngine.getTuningJobDefinitionsForParameterSuggestion()
+        .eq(TuningJobDefinition.TABLE.tuningAlgorithm + "." + TuningAlgorithm.TABLE.optimizationAlgo,
+            TuningAlgorithm.OptimizationAlgo.HBT.name())
+        .findList();
+
+    logger.debug(" Number of Total Jobs " + this._executionEngine + " " + totalJobs.size());
+    return totalJobs;
+  }
+
+  @Override
+  protected void saveJobState(JobTuningInfo jobTuningInfo, JobDefinition job) {
+    jobTuningInfo.setTunerState("{}");
+  }
+
+  @Override
+  public JobTuningInfo generateParamSet(JobTuningInfo jobTuningInfo) {
+    logger.debug("Generating param set for job: " + jobTuningInfo.getTuningJob().jobName);
+    String newTunedParameters = generateParamSet(jobTuningInfo.getParametersToTune(), jobTuningInfo.getTuningJob());
+    jobTuningInfo.setTunerState(newTunedParameters);
+    return jobTuningInfo;
+  }
+
+  private String generateParamSet(List<TuningParameter> tuningParameters, JobDefinition job) {
+    JobExecution jobExecution = JobExecution.find.select("*")
+        .where()
+        .eq(JobExecution.TABLE.job, job)
+        .order()
+        .desc(JobExecution.TABLE.updatedTs)
+        .setMaxRows(1)
+        .findUnique();
+    logger.debug("Job Status " + jobExecution.executionState.name());
+    if (jobExecution.executionState.name().equals(JobExecution.ExecutionState.IN_PROGRESS.name())
+        || jobExecution.executionState.name().equals(JobExecution.ExecutionState.NOT_STARTED.name())) {
+      logger.debug(" Job is still running , cannot use for param generation ");
+      return "";
+    }
+
+    List<AppResult> results = getAppResults(jobExecution);
+    if (results == null || results.size() == 0) {
+      logger.debug(
+          " Job is analyzing  , cannot use for param generation " + jobExecution.id + " " + jobExecution.job.id);
+      return "";
+    }
+    String idParameters = null;
+    try {
+      idParameters = this._executionEngine.parameterGenerationsHBT(results, tuningParameters);
+      if (idParameters == null) {
+        logger.error(" id parameters are null ");
+        idParameters = "";
+      }
+    } catch (Exception e) {
+      logger.error("Exception in getting specific parameters ", e);
+    }
+    return idParameters.toString();
```

Comment:
If we are expecting that parameterGenerationsHBT may throw a runtime exception(as we are catching it), this line can throw NPE.

Reply:
changed in https://github.com/linkedin/dr-elephant/pull/450